### DISCUSSION TITLE
External Viewers broken

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#3900] [General]
As of recentlyish, the External Viewers are broken. I waited a bit thinking it would get immediately spotted. I'm using 3.6.15899 64-bit pre-rel (the latest).
It seems the file that's sent to the external viewer has been broken on non-alphanumeric characters such as space or dash. If it lacks them it works, add them in and the external viewer fails because it is passed an incorrect file. There are also issues when the external viewer command line itself has such characters in. So,

This launches but now fails to open the file when the file it is passed has such characters in -
CommandLine = "C:\Program Files\SumatraPDF\SumatraPDF.exe" -page %p "%1"

And now this won't even run due to the "right-clicks" part of the command -
CommandLine = "C:\Windows\System32\wscriptcopy.exe" "D:\Scripts\right-clicks\locator.vbs" "%1"

David

**Comment by GitHubRulesOK** [comment]
there were very recent changes to the use of command line "quotes"
SO I too find they are not what the were longer time ago 
only the current pre-release behaviour can be changed so ensure you are comparing current behaviour

generally I find that quoting the quotes helps so `"""` seemed to work in 3.5

However most recent 3.6 pre-rel copy seems to be sending all sorts of off mixes so using your call sumatrapdf via cmd then the filename is sent first and triggers Acrobat !!
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/805fe2ee-6910-4f96-b55f-981c6f647b39)

**Reply by GitHubRulesOK** [reply]
elevated to issues https://github.com/sumatrapdfreader/sumatrapdf/issues/3901
